CarpetX/src/io_openpmd.cxx

```diff
@@ -1206,8 +1206,8 @@ void carpetx_openpmd_t::InputOpenPMD(const cGH *const cctkGH,
                 constexpr std::uint64_t ipoison =
                     0xfff8000000000000ULL + 0xdeadbeef;
 #endif
-                char poison[CCTK_VarTypeSize(cgroup.vartype)];
-                std::memcpy(&poison, &ipoison, sizeof(poison));
+                std::vector<char> poison(CCTK_VarTypeSize(cgroup.vartype));
+                std::memcpy(poison.data(), &ipoison, poison.size());
```

Comment:
This won't work when `poison.size()` is larger than `sizeof ipoison`.

Maybe define a few poison value for various types instead?

Reply:
Well it'll copy in something but leave some bits unitialized. 

Right now poison is actually a NaN.

`valid.cxx` instead uses:  
 

```

  const size_t typesize = size_t(CCTK_VarTypeSize(group.vartype));
  char poison[typesize];
  // for int: deadbeef for little endian machines
  for (size_t i = 0; i < typesize; i += sizeof poison)
    std::memcpy(&poison[i], &ipoison, std::min(sizeof poison, typesize - i));
```

which should actually fail with nvcc and `-std=c++17` since variable size arrays are not in C++ (and I got a failure b/c of that in `io_openpmd.cxx` ).  
  
Should we use the same logic as in `valid.cxx` also in `io_openpmd.cxx`?

Reply:
added like this in the current version